samples/Microsoft.Bot.Samples.Ai.Luis/Controllers/MessagesController.cs

```diff
@@ -2,52 +2,58 @@
+using System.Threading.Tasks;
 
-namespace Microsoft.Bot.Samples.Ai.Luis
+namespace Microsoft.Bot.Samples.Ai.Luis.Controllers
 {
     [Route("api/[controller]")]
     public class MessagesController : Controller
     {
         static BotFrameworkAdapter adapter;
-
-        /// <summary>
-        /// In this sample Bot, a new instance of the Bot is created by the controller 
-        /// on every incoming HTTP reques. The bot is constructed using the credentials
-        /// found in the config file. Note that no credentials are needed if testing
-        /// the bot locally using the emulator. 
-        /// </summary>        
+    
         public MessagesController(IConfiguration configuration)
         {
             if (adapter == null)
             {
-                adapter = new BotFrameworkAdapter(configuration)
-                    .Use(new LuisRecognizerMiddleware("xxxxxx", "xxxxxx"));
-
-                // LUIS with correct baseUri format example
-                //.Use(new LuisRecognizerMiddleware("xxxxxx", "xxxxxx", "https://xxxxxx.api.cognitive.microsoft.com/luis/v2.0/apps"))
-
+                adapter = new BotFrameworkAdapter(new ConfigurationCredentialProvider(configuration))
+                    .Use(new BatchOutputMiddleware())
+                    .Use(new LuisRecognizerMiddleware(new LuisModel("modelId", "subscriptionKey", new Uri("https://xxxxxx.api.cognitive.microsoft.com/luis/v2.0/apps/"))
+                    // If you want to get all intents scorings, add verbose in luisOptions
+                    // .Use(new LuisRecognizerMiddleware(new LuisModel("modelId", "subscriptionKey", new Uri("https://xxxxxx.api.cognitive.microsoft.com/luis/v2.0/apps/")), null, luisOptions: new LuisRequest { Verbose = true }
+                    ));
             }
         }
 
         private Task BotReceiveHandler(IBotContext context)
         {
             if (context.Request.Type == ActivityTypes.Message)
             {
-                context.Reply($"the top intent was: {context.TopIntent.Name}");
+                var luisResult = context.Get<RecognizerResult>(LuisRecognizerMiddleware.LuisRecognizerResultKey);
 
-                foreach (var entity in context.TopIntent.Entities)
+                if (luisResult != null)
                 {
-                    context.Reply($"entity: {entity.ValueAs<string>()}");
+                    (string key, double score) topItem = luisResult.GetTopScoringIntent();
+                    context.Batch().Reply($"The **top intent** was: **'{topItem.key}'**, with score **{topItem.score}**");
+
+                    context.Batch().Reply($"Detail of intents scorings:");
+                    var intentsResult = new List<string>();
+                    foreach (var intent in luisResult.Intents)
+                    {
+                        intentsResult.Add($"* '{intent.Key}', score {intent.Value}");
+                    }
+                    context.Batch().Reply(string.Join("\n\n", intentsResult));
```

Comment:
That's the thinking behind adding two \n to the reply?

Reply:
In addition with the markdown in `$"* '{intent.Key}', score {intent.Value}"` formatting of each intent, it will display all the intents as an unordered list.
Doubling the `\n` is necessary for the bot to display a new line, if there is only one a new line will not be present.